### PR TITLE
AS-365, AS-370: support for data repo snapshot row counts and primary keys [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
for entity-type-metadata endpoint, honor the row counts and primary keys returned by data repo.

### CODE REVIEW

File: core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProviderSpec.scala [hunk 1]
```diff
@@ -13,39 +14,34 @@ class DataRepoEntityProviderSpec extends AsyncFlatSpec with DataRepoEntityProvid
...
-      dataRepoDAO = new SpecDataRepoDAO(Right( createSnapshotModel( List( ("foo", List.empty[String]), ("bar", List("one", "two")) ) ) )))
-
-    val expected = Map(
-      ("foo", EntityTypeMetadata(-1, "", Seq.empty[String])),
-      ("bar", EntityTypeMetadata(-1, "", Seq("one", "two"))))
-
-    provider.entityTypeMetadata() map { metadata: Map[String, EntityTypeMetadata] =>
-      assertResult(expected) { metadata }
-    }
-  }
```
Comment (davidangb) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityProviderSpec.scala:47]: removed this test, as it is now incorporated into the `"return entity type metadata in the golden path"` test

File: project/Dependencies.scala [hunk 2]
```diff
@@ -94,7 +94,7 @@ object Dependencies {
   val accessContextManager = "com.google.apis" % "google-api-services-accesscontextmanager" % "v1beta-rev55-1.25.0"
 
   val workspaceManager = excludeGuavaJDK5("bio.terra" % "terra-workspace-manager-client" % "0.0.2-SNAPSHOT")
-  val dataRepo = excludeGuavaJDK5("bio.terra" % "datarepo-client" % "1.0.1-SNAPSHOT").exclude("javax.ws.rs","javax.ws.rs-api")
+  val dataRepo = excludeGuavaJDK5("bio.terra" % "datarepo-client" % "1.0.2-SNAPSHOT").exclude("javax.ws.rs","javax.ws.rs-api")
```
Comment (davidangb) [inline project/Dependencies.scala]: necessary, as it adds support for `rowCount` on the table model

**Comment by davidangb** [comment]
rawls-build failed connecting to bintray; should be transient. Jenkins retest.

**Comment by davidangb** [comment]
welp. not so transient; still having bintray issues. one more time? Jenkins retest

File: project/Dependencies.scala [hunk 3]
```diff
@@ -95,7 +95,7 @@ object Dependencies {
   val accessContextManager = "com.google.apis" % "google-api-services-accesscontextmanager" % "v1beta-rev55-1.25.0"
 
   val workspaceManager = excludeGuavaJDK5("bio.terra" % "terra-workspace-manager-client" % "0.0.2-SNAPSHOT")
-  val dataRepo = excludeGuavaJDK5("bio.terra" % "datarepo-client" % "1.0.1-SNAPSHOT").exclude("javax.ws.rs","javax.ws.rs-api")
+  val dataRepo = excludeGuavaJDK5("bio.terra" % "datarepo-client" % "1.0.3-SNAPSHOT")
```
Comment (davidangb) [inline project/Dependencies.scala:98]: DataBiosphere/jade-data-repo#499 removed the need to exclude `javax.ws.rs-api`

**Comment by davidangb** [comment]
failure registering test users during fiab-start `oauth2client.client.HttpAccessTokenRefreshError: internal_failure` ... jenkins retest

**Review by ansingh7115 — APPROVED** [review]

**Pull request merged**